packages/common/src/stores/AccountController.ts

```diff
@@ -509,14 +508,17 @@ export default class AccountController {
     return this.features;
   }
 
-  private async afterLogin(user: Customer, customerConsents: CustomerConsent[] | null, accessModel: string, shouldSubscriptionReload: boolean = true) {
+  private async afterLogin(user: Customer, customerConsents: CustomerConsent[] | null, shouldReloadSubscription = true) {
     useAccountStore.setState({
       user,
       customerConsents,
     });
 
+    const { accessModel } = useConfigStore.getState();
+
     await Promise.allSettled([
-      accessModel === ACCESS_MODEL.SVOD && shouldSubscriptionReload ? this.reloadActiveSubscription() : Promise.resolve(),
+      shouldReloadSubscription && accessModel === 'SVOD' ? this.reloadActiveSubscription() : Promise.resolve(),
+      shouldReloadSubscription && accessModel !== 'SVOD' ? this.refreshEntitlements?.() : Promise.resolve(), // For non-SVOD platforms, there could be TVOD items
```

Comment:
```suggestion
      // For SVOD platforms, we reload the subscriptions and entitlements
      shouldReloadSubscription && accessModel === 'SVOD' ? this.reloadActiveSubscription() : this.refreshEntitlements?.(),
      // For non-SVOD platforms, there could be TVOD items, so we only the reload entitlements
      shouldReloadSubscription && accessModel !== 'SVOD' ? this.refreshEntitlements?.() : Promise.resolve(),
```

Reply:
Perhaps, we can rename `reloadActiveSubscription` to `reloadSubscriptions` and make let it decide if it needs to refresh SVOD subscriptions.

Then we can just call:

```suggestion
      shouldReloadSubscription ? this.reloadSubscriptions() : Promise.resolve(),
```

And check the AccessModel in the `reloadSubscriptions` function.

Reply:
Great suggestion!